File: client/src/app/_models/livro.ts
import { User } from './user';

export interface Livro {
  id: number;
  fimLeitura: Date | null;
  inicioLeitura: Date | null;
  nome: string;
  urlCapa: string | null;
  autor: string;
  resumo: string | null;
  appUser: User | null;
  paginas: number | null;
  categoria: string | null;
}

export interface LivroDto {
  id: number;
  fimLeitura: Date | null;
  inicioLeitura: Date | null;
  nome: string;
  urlCapa: string | null;
  autor: string;
  resumo: string | null;
  appUserId: string | null;
  appUser: string | null;
  paginas: number | null;
  categoria: string | null;
}

File: API/Entities/Livro.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities;

public class Livro
{

	public int Id { get; set; }
	[StringLength(150)]
	public required string Nome { get; set; }

	[StringLength(150)]
	public required string Autor { get; set; }
	public DateTime? FimLeitura { get; set; }

	public DateTime? InicioLeitura { get; set; }

	[StringLength(300)]
	public string? UrlCapa { get; set; }

	[StringLength(500)]
	public string? Resumo { get; set; }

	public required string AppUserId { get; set; }

	[ForeignKey("AppUserId")]
	public AppUser? AppUser { get; set; }



}

public class LivroMock
{
	public int Id { get; set; }
	public string? Nome { get; set; }
	public string? Autor { get; set; }
	public string? Data { get; set; }
}
